src/Modules/UngradedModules.jl

```diff
@@ -0,0 +1,3653 @@
+    end
+  end
+
+  new_generators = delete_rows(new_generators, to_delete)
+
+  if length(new_generators)==0
+    zero_module = FreeMod(R,0)
+    injection = FreeModuleHom(zero_module, M, [])
+    projection = SubQuoHom(M, zero_module, [zero(zero_module) for i=1:ngens(M)])
+    # TODO early return or register morphisms?
+    return zero_module,injection,projection
+  else
+    SQ = iszero(new_relations) ? SubQuo(SubModuleOfFreeModule(new_generators)) : SubQuo(new_generators, new_relations)
+    injection = SubQuoHom(SQ, M, injection_matrix)
+    projection = SubQuoHom(M, SQ, projection_matrix[:,1:size(projection_matrix)[2]-size(new_relations)[1]])
+  end
+  register_morphism!(injection)
+  register_morphism!(projection)
+  injection.inverse_isomorphism = projection
+  projection.inverse_isomorphism = injection
+
+  return SQ,injection,projection
+end
+
+######################################
+# Not only for testing
+######################################
+@doc Markdown.doc"""
+    map(F::FreeMod{T}, A::MatElem{T}) where T
+
+Converts a given $n \times m$-matrix into the corresponding morphism $A : R^n \to F$, 
+with `rank(F) == m`.
+"""
+function map(F::FreeMod{T}, A::MatElem{T}) where T
+  R = base_ring(F)
+  F_domain = FreeMod(R, nrows(A))
+
+  phi = FreeModuleHom(F_domain, F, A)
+  return phi
+end
+
+@doc Markdown.doc"""
+    map(A::MatElem)
+
+Converts a given $n \times m$-matrix into the corresponding morphism $A : R^n \to R^m$.
+"""
+function map(A::MatElem)
+  R = base_ring(A)
+  F_codomain = FreeMod(R, ncols(A))
+  return map(F_codomain,A)
+end
+
+@doc Markdown.doc"""
+    isinjective(f::ModuleMap)
+
+Test if `f` is injective.
+"""
+function isinjective(f::ModuleMap)
+  return iszero(kernel(f)[1])
+end
+
+@doc Markdown.doc"""
+    issurjective(f::ModuleMap)
+
+Test if `f` is surjective.
+"""
+function issurjective(f::ModuleMap)
+  return image(f)[1] == codomain(f)
+end
+
+@doc Markdown.doc"""
+    isbijective(f::ModuleMap)
+
+Test if `f` is bijective.
+"""
+function isbijective(f::ModuleMap)
+  return isinjective(f) && issurjective(f)
+end
+
+@doc Markdown.doc"""
+    iswelldefined(H::ModuleMap)
+
+Test if `H` is well-defined.
+"""
+function iswelldefined(H::ModuleMap)
+  if typeof(H) <: FreeModuleHom
+    return true
+  end
+  M = domain(H)
+  C = matrix(present_as_cokernel(M).quo)
+  n,m = size(C)
+  g = M[1]
+  ImH = map(x -> H(x), gens(M))
+  for i=1:n
+    if !iszero(Base.sum([C[i,j]*ImH[j] for j=1:m]))
+      return false
+    end
+  end
+  return true
+end
+
+#############################################
+# Test hom
+#############################################
+function to_julia_matrix(A::Union{MatElem})
+  return eltype(A)[A[i, j] for i = 1:nrows(A), j = 1:ncols(A)]
+end
+
+function Base.reshape(M::MatElem, n, m)
+  julia_matrix = to_julia_matrix(M)
+  julia_matrix = reshape(julia_matrix, n, m)
+  R = base_ring(M)
+  mat_space = MatrixSpace(R, n, m)
+  return mat_space(julia_matrix)
```

Comment:
I think this should also work (untested)
```suggestion
  return typeof(M)(n, m, julia_matrix)
```